spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/LoggingWebMvcConfigurerSettings.java

```diff
@@ -0,0 +1,51 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+/**
+ * Holds settings for @{@link LoggingWebMvcConfigurer}
+ * @author Chengyuan Zhao
+ */
+public class LoggingWebMvcConfigurerSettings {
```

Comment:
That's a bit awkward solution. I would stay away of such an `enum` simulation.
We have `TraceIdExtractor` strategy and some out-of-the box implementations. 
We should just allow end-user to configure his/her own `TraceIdExtractor` bean, meanwhile we should be ready for its injection.

Moreover right now you fully don't allow to inject any custom `TraceIdExtractor`.
That is really much cleaner to just let to configure beans and provide some default when nothing is provided.

Reply:
Yes I agree that sounds better. I will make changes to make it more like that.